astropy/stats/sigma_clipping.py

```diff
@@ -161,25 +175,51 @@ def _sigma_clip(data, sigma=3, sigma_lower=None, sigma_upper=None, iters=5,
     Note that along the other axis, no points would be masked, as the
     variance is higher.
     """
+    def perform_clip(_filtered_data, _kwargs):
+        """
+        Perform sigma clip by comparing the data to the minimum and maximum
+        values (median + sig * standard deviation). Use sigma_lower and
+        sigma_upper to get the correct limits. Data values less or greater
+        than the minimum / maximum values will have True set in the mask array.
+        """
+        max_value = cenfunc(_filtered_data, **_kwargs)
+        std = stdfunc(_filtered_data, **_kwargs)
+        min_value = max_value - std * sigma_lower
+        max_value += std * sigma_upper
+        if axis is not None:
+            if axis > 0:
+                min_value = np.expand_dims(min_value, axis=axis)
+                max_value = np.expand_dims(max_value, axis=axis)
+        _filtered_data.mask |= _filtered_data > max_value
+        _filtered_data.mask |= _filtered_data < min_value
 
     if sigma_lower is None:
         sigma_lower = sigma
     if sigma_upper is None:
         sigma_upper = sigma
 
+    kwargs = dict()
+
     if axis is not None:
-        cenfunc_in = cenfunc
-        stdfunc_in = stdfunc
-        cenfunc = lambda d: np.expand_dims(cenfunc_in(d, axis=axis),
-                                           axis=axis)
-        stdfunc = lambda d: np.expand_dims(stdfunc_in(d, axis=axis),
-                                           axis=axis)
+        kwargs['axis'] = axis
 
     if np.any(~np.isfinite(data)):
         data = np.ma.masked_invalid(data)
         warnings.warn("Input data contains invalid values (NaNs or infs), "
                       "which were automatically masked.", AstropyUserWarning)
 
+    if masked is False and HAS_SCIPY is True:
```

Comment:
There is one other problem here. ``scipy.stats.sigmaclip`` does not allow for an ``axis`` parameter but we do. So maybe wrap this ``if ... elif ...`` in an 
```
if axis is None: 
   <fill in the current if ... elif ...>
elif not masked: 
    warnings.warn("You specified an axis parameter so we need to operate on masked data", 
                            AstropyUserWarning)
```
Additionally something similar should happen with the ``iters`` parameter - or you simply wrap the call to scipys function inside an ``for i in range(iters):`` loop before returning.